pkg/ds/stack/stack.go

```diff
@@ -0,0 +1,52 @@
+package stack
+
+// Stack is a generic stack.
+type Stack[T any] struct {
+	elems []T
+}
+
+// New returns a new instance of Stack.
+// hint: hint for stack size.
+func New[T any](hint int) *Stack[T] {
+	if hint < 0 {
+		hint = 0
+	}
+
+	return &Stack[T]{elems: make([]T, 0, hint)}
+}
+
+// Empty returns true if the stack is empty.
+func (s *Stack[T]) Empty() bool {
+	return len(s.elems) == 0
+}
+
+// Len returns number of elements in the stack.
+func (s *Stack[T]) Len() int {
+	return len(s.elems)
+}
+
+// Peek views the top element of the stack.
+// nolint: ireturn
+func (s *Stack[T]) Peek() T {
+	return s.elems[len(s.elems)-1]
+}
+
+// Push pushes a new element into the stack.
+func (s *Stack[T]) Push(elem T) {
+	s.elems = append(s.elems, elem)
+}
+
+// Pop pops the top element from the stack.
+// Panic if the stack is empty.
+// nolint: ireturn
+func (s *Stack[T]) Pop() T {
```

Comment:
Shouldn't panic. Should return error, user will decide panic or error handle

Reply:
I thought about it, not sure if it better to return error for leave the check for the caller.